packages/grid/test/data-provider.common.js

```diff
@@ -203,6 +203,25 @@ describe('data provider', () => {
     it('should set itemHasChildren path by default', () => {
       expect(grid.itemHasChildrenPath).to.equal('children');
     });
+
+    it('should request items when sorting is applied initially', () => {
+      // Initialize a fresh grid
+      grid.remove();
+      grid = fixtureSync(`
+        <vaadin-grid>
+          <vaadin-grid-sort-column path="value" direction="asc"></vaadin-grid-sort-column>
+        </vaadin-grid>
+      `);
+
+      // Set data provider
+      grid.dataProvider = (params, callback) => {
+        infiniteDataProvider(params, (items) => callback(items, 1));
```

Comment:
```suggestion
        callback(['Item'], 1);
```

Reply:
Good point, simplified the data provider